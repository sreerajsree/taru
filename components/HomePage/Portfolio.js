import PortfolioItem from './PortfolioItem';

export default function Portfolio({ articles }) {
  const [newestArticle] = articles;
  return (
    <section>
      <div className="flex flex-col items-center w-full mt-16 md:mt-32 mb-0 md:mb-16 font-title">
        <h2 className="text-7xl mb-6 md:mb-10">Portfolio</h2>
        {/* <p className="text-lg my-8">Past. Present. Future.</p> */}
        {[newestArticle].map(({
          title,
          image,
          date,
          description,
          excerpt,
        }, index) => (
          <PortfolioItem
            key={`portfolio-item-${title}`}
            invert={index % 2 === 1}
            title={title}
            image={image}
            date={date}
            description={description}
            excerpt={excerpt}
          />
        ))}
      </div>
    </section>
  );
}
