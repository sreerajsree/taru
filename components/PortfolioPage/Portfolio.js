import { useEffect, useState } from 'react';
import PortfolioItem from './PortfolioItem';
import { calculateNumItemsOnRow } from '../../lib/utils';

export default function Portfolio({ articles }) {
  const [numItemsOnRow, setNumItemsOnRow] = useState(0);

  useEffect(() => {
    const num = calculateNumItemsOnRow(window.innerWidth);
    setNumItemsOnRow(num);
  }, []);
  return (
    <section>
      <div className="flex flex-row flex-wrap justify-center flex-1">
        {articles.map(({
          image,
          title,
          date,
          description,
          excerpt,
          slug,
        }, index) => {
          const onTopRow = index < numItemsOnRow;
          return (
            <PortfolioItem
              key={`portfolio-item-${title}`}
              onTopRow={onTopRow}
              title={title}
              image={image}
              date={date}
              description={description}
              excerpt={excerpt}
              slug={slug}
            />
          );
        })}
      </div>
    </section>
  );
}
