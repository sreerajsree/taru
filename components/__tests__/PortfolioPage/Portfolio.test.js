import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as nextRouter from 'next/router';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

import Portfolio from '../../PortfolioPage/Portfolio';

jest.mock('react-use');
nextRouter.useRouter = jest.fn();

nextRouter.useRouter.mockImplementation(() => ({ route: '/', query: {} }));

const article = {
  image: 'image.jpg',
  title: 'title',
  date: '2019-10-10T00:00:00',
  description: 'desc',
  excerpt: 'excerpt',
  slug: 'slug',
  onTopRow: true,
};

const articles = [
  article,
  {
    ...article,
    title: 'title2',
  },
];

describe('Hero', () => {
  beforeEach(() => {
    mockAllIsIntersecting(true);
  });
  test('renders and matches snapshot', () => {
    const { asFragment } = render(<Portfolio articles={articles} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check that the title, description and date are correctly rendered', () => {
    render(<Portfolio articles={[article]} />);
    expect(screen.getByText('title')).toBeInTheDocument();
    expect(screen.getByText('desc')).toBeInTheDocument();
    expect(screen.getByText('October, 2019')).toBeInTheDocument();
  });
});
