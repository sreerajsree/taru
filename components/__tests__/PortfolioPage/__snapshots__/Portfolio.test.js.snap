// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders and matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="flex flex-row flex-wrap justify-center flex-1"
    >
      <div
        class="flex flex-col xl:w-1/4 lg:w-33% md:w-50% p-1"
      >
        <div
          class="flex flex-col h-full"
          style="opacity: 0; transform: translateY(300px) translateZ(0);"
        >
          <div
            class="bg-center bg-cover h-50vh relative"
            style="background-image: url(test-file-stub);"
          >
            <div
              class="absolute top-0 left-0 bg-img-tint h-full w-full flex justify-center items-center"
            >
              <h2
                class="text-2xl p-2 md:px-16 font-title text-center text-white"
              >
                title
              </h2>
            </div>
          </div>
          <div
            class="w-full flex items-center justify-top flex-col text-center bg-grey-custom pb-12 md:pb-32 flex-1 relative"
          >
            <p
              class="text-base md:text-lg font-title pb-2 pt-4 md:pt-6 md:pb-4"
            >
              <time
                datetime="2019-10-10T00:00:00"
              >
                October, 2019
              </time>
            </p>
            <p
              class="text-base md:text-lg font-body py-2 md:py-3 px-4 md:px-6 lg:px-8"
            >
              desc
            </p>
            <p
              class="text-base md:text-lg font-body py-2 md:py-3 px-4 md:px-6 lg:px-8"
            >
              'excerpt'
            </p>
            <a
              class="px-4 py-2 mt-6 md:mt-8 border-solid border border-black font-body text-base ease-in-out transition duration-500 md:absolute md:bottom-0 md:mb-16 border-pink-custom text-pink-custom hover:text-grey-custom hover:bg-pink-custom"
              href="/portfolio/slug"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col xl:w-1/4 lg:w-33% md:w-50% p-1"
      >
        <div
          class="flex flex-col h-full"
          style="opacity: 0; transform: translateY(300px) translateZ(0);"
        >
          <div
            class="bg-center bg-cover h-50vh relative"
            style="background-image: url(test-file-stub);"
          >
            <div
              class="absolute top-0 left-0 bg-img-tint h-full w-full flex justify-center items-center"
            >
              <h2
                class="text-2xl p-2 md:px-16 font-title text-center text-white"
              >
                title2
              </h2>
            </div>
          </div>
          <div
            class="w-full flex items-center justify-top flex-col text-center bg-grey-custom pb-12 md:pb-32 flex-1 relative"
          >
            <p
              class="text-base md:text-lg font-title pb-2 pt-4 md:pt-6 md:pb-4"
            >
              <time
                datetime="2019-10-10T00:00:00"
              >
                October, 2019
              </time>
            </p>
            <p
              class="text-base md:text-lg font-body py-2 md:py-3 px-4 md:px-6 lg:px-8"
            >
              desc
            </p>
            <p
              class="text-base md:text-lg font-body py-2 md:py-3 px-4 md:px-6 lg:px-8"
            >
              'excerpt'
            </p>
            <a
              class="px-4 py-2 mt-6 md:mt-8 border-solid border border-black font-body text-base ease-in-out transition duration-500 md:absolute md:bottom-0 md:mb-16 border-pink-custom text-pink-custom hover:text-grey-custom hover:bg-pink-custom"
              href="/portfolio/slug"
            >
              Read more
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
